src/main/java/fi/aalto/cs/apluscourses/ui/exercise/ExercisesView.java

```diff
@@ -44,6 +48,14 @@ private void createUIComponents() {
     exerciseGroupsTree.setCellRenderer(new ExercisesTreeRenderer());
     exerciseGroupsTree.addNodeAppliedListener(
         ActionUtil.createOnEventLauncher(OpenItemAction.ACTION_ID, exerciseGroupsTree));
+
+    new TreeSpeedSearch(exerciseGroupsTree, x -> {
+      Object treeObject = ((DefaultMutableTreeNode) x.getLastPathComponent()).getUserObject();
+      if (treeObject instanceof ExerciseViewModel) {
+        return ((ExerciseViewModel) treeObject).getPresentableName();
+      }
+      return ""; // disable searching on entries other than assignments
```

Comment:
I guess searching for submissions is pretty pointless, since they are all named "Submission x", but could searching for rounds/weeks be useful?

Reply:
Aren't all weeks called "week/round X" as well? There's no problem in adding that of course, but exercises are special in the way that they have custom names rather than just numbers

Reply:
And all weeks are instantly visible & sorted when launching IntelliJ, so it's arguably faster to just locate the desired week with your eyes

Reply:
Well, to be specific, weeks/rounds are called whatever the course calls them. Submissions however are named by us. The difference with rounds and submissions is that submissions will have duplicate names ("Submission 2") might have 10 matches, but ("Week 2") should usually only have one match.

But I think you're right, it's probably not very useful in practice to search for them.